docs/how-to-deploy-and-call-a-contract/call-deployed.md

```diff
@@ -6,39 +6,43 @@ sidebar_position: 5
 
 ## Overview
 In this tutorial, we will interact with a deployed smart contract by calling its public method, in a separate process or by a different party.
-We need to create an instance corresponding to the deployed contract on chain.
+
+To do this, we need to create a smart contract instance that corresponds to the deployed contract on chain.
 
 ## The Smart Contract
 
-We will reuse [the `Counter` contract](../how-to-write-a-contract/stateful-contract.md#create-a-stateful-contract).
+We will reuse the stateful `Counter` contract [from a previous step](../how-to-write-a-contract/stateful-contract#create-a-stateful-contract).
 
 ```ts
 export class Counter extends SmartContract {
-
-    @prop(true)
-    count: bigint
-
-    constructor(count: bigint) {
-        super(...arguments)
-        this.count = count
-    }
-
-    @method()
-    public incrementOnChain() {
-        // Increment counter.
-        this.increment()
-
-        // Ensure next output will contain this contracts code with
-        // the updated count property.
-        const amount: bigint = this.ctx.utxo.value
-        const outputs: ByteString = this.buildStateOutput(amount) + this.buildChangeOutput()
-        assert(this.ctx.hashOutputs == hash256(outputs), 'hashOutputs mismatch')
-    }
-
-    @method()
-    increment(): void {
-        this.count++
-    }
+  // stateful
```

Comment:
made the code examples between the 2 files line-for-line consistent.

maybe there's a way to extract repeated code samples into something like a "partial" or "include"